Modules/+Drivers/+Kinesis/KinesisBSC203.m

```diff
@@ -0,0 +1,347 @@
+        function connect(obj, serialNo) % serialNo := str
+            obj.GetDevices;  % Call this to build device list if not already done
+            if ~obj.isconnected()   % Connect and initialize device if not connected  
+                obj.deviceNET = Thorlabs.MotionControl.Benchtop.StepperMotorCLI.BenchtopStepperMotor.CreateBenchtopStepperMotor(serialNo);  % Create an instance of .NET BenchtopStepperMotor
+                obj.deviceNET.Connect(serialNo);    % Connect to device via .NET interface
+                for i = obj.motor_channels
+                    if ~isnan(i)
+                        obj.channelsNET{i} = obj.deviceNET.GetChannel(i);   % Get channel objects of the device
+                        obj.channelsNET{i}.ClearDeviceExceptions(); % Clear device exceptions via .NET interface
+                    else
+                        disp('Motor channel is empty')
+                    end
+                end
+                                                                        
+                obj.initialize(serialNo)    % Initialize the device
+            else    % Device already connected
+                error('Device is already connected')
+            end
+            obj.updatestatus   % Update status variables from device
+        end
+
+        function initialize(obj, serialNo) % Initialize all three channels of the device, serialNo := str
+            for i = obj.motor_channels
+                if ~isnan(i)
+                    try
+                        if ~obj.channelsNET{i}.IsSettingsInitialized() 
+                            obj.channelsNET{i}.WaitForSettingsInitialized(obj.TIMEOUTSETTINGS); % Initialize the ith channel
+                        else
+                            disp('Device Already Initialized.')
+                        end
+                        if ~obj.channelsNET{i}.IsSettingsInitialized() % Device not successfully initialized
+                            error('Unable to initialize device')
+                        end
+                        obj.channelsNET{i}.StartPolling(obj.TPOLLING);  % Start polling device via .NET interface
+                        obj.channelsNET{i}.EnableDevice();  % Enable device via .NET interface
+
+                        % Initialize motor configuration
+                        deviceID = obj.channelsNET{i}.DeviceID;
+                        settingsLoadOption = Drivers.Kinesis.Kinesis_invisible.GetSettingsLoadOption(serialNo, deviceID);
+                        obj.motorSettingsNET{i} = obj.channelsNET{i}.GetMotorConfiguration(serialNo, settingsLoadOption);
+
+                        % Initialize current motor settings
+                        obj.currentDeviceSettingsNET{i}=obj.channelsNET{i}.MotorDeviceSettings;
+                        obj.deviceInfoNET{i} = obj.channelsNET{i}.GetDeviceInfo();  % Get deviceInfo via .NET interface
+                    catch
+                        error(['Unable to initialize channel ', num2str(i)]);
```

Comment:
Is there anything helpful in the caught error that should be included in the rethrown error?

Reply:
Will include a line to rethrow the caught error